infra/docker/realtime/prod/Dockerfile

```diff
@@ -1,36 +1,38 @@
-FROM node:16.18.1-alpine AS builder
+FROM node:16.18.1-alpine AS base
 
 WORKDIR /app
 
-# Ensure that the node module layer can be cached
-COPY package.json .
-COPY yarn.lock .
-RUN yarn install --frozen-lockfile --production --ignore-scripts --prefer-offline
+RUN yarn global add turbo
+COPY . .
+RUN turbo prune --scope='@apps/hash-realtime' --docker
 
-# Also ensure that api node modules can be cached
-COPY apps/hash-realtime/package.json apps/hash-realtime/
-COPY libs/@local/eslint-config/package.json libs/@local/eslint-config/
-COPY libs/@local/hash-backend-utils/package.json libs/@local/hash-backend-utils/
-COPY libs/@local/tsconfig/package.json libs/@local/tsconfig/
-RUN yarn workspace @apps/hash-realtime install --frozen-lockfile --ignore-scripts --prefer-offline
+FROM node:16.18.1-alpine as installer
 
-COPY libs/@local/eslint-config libs/@local/eslint-config
-COPY libs/@local/hash-backend-utils libs/@local/hash-backend-utils
-COPY libs/@local/tsconfig libs/@local/tsconfig
-COPY apps/hash-realtime apps/hash-realtime
 
-#########################################################################################
+WORKDIR /usr/local/src/
 
-FROM node:16.18.1-alpine
+COPY --from=base /app/out/json/ .
+COPY --from=base /app/out/yarn.lock ./yarn.lock
+COPY --from=base /app/out/full/turbo.json turbo.json
 
-# Run as a non-root user
-RUN addgroup -S appgroup && adduser -S appuser -G appgroup
-COPY --from=builder --chown=appgroup:appuser /app /app
+RUN yarn install --frozen-lockfile --prefer-offline \
```

Comment:
Please note, that this won't run `codegen` anymore. You may want to consider moving the `COPY` command of `turbo.json` below this and run `RUN yarn turbo codegen --filter '@apps/hash-realtime'`

Reply:
Oh yes, thank you. Even though I think this doesn't need codegen, I will still add it :+1:

Reply:
Maybe it's even sensible to call `build` here, but I'm unsure about that.

Reply:
Yeah I think that's better, should have codegen as a dep I suppose

Reply:
yes, that dependency is global unless overwritten 🙂

Reply:
`turbo.json` seems to be required for the `yarn install` for now (fixed in #2463 perhaps?) do we want to wait for that to merge, or would the requirement be there regardless?

Reply:
Let's wait for #2463, this will remove the dependency. 🙂